include/itkTileMontage.hxx

```diff
@@ -438,198 +440,221 @@ TileMontage<TImageType, TCoordinate>
+TileMontage<TImageType, TCoordinate>
+::ResampleSingleRegion(unsigned i)
+{
   bool interpolate = true;
   if (m_PCMOptimizer->GetPeakInterpolationMethod() == PCMOptimizerType::PeakInterpolationMethod::None)
     {
     interpolate = false;
     }
-  for (unsigned i = 0; i < regions.size(); i++) //TODO: parallelize this loop!
-    {
-    ImageRegionIteratorWithIndex<ImageType> oIt(result, regions[i]);
+  ImageRegionIteratorWithIndex<ImageType> oIt(m_SingleImage, m_Regions[i]);
 
-    if (regionContributors[i].empty()) //not covered by any tile
+  if (m_RegionContributors[i].empty()) //not covered by any tile
+    {
+    while (!oIt.IsAtEnd())
       {
-      while (!oIt.IsAtEnd())
-        {
-        oIt.Set(background);
-        ++oIt;
-        }
+      oIt.Set(m_Background);
+      ++oIt;
+      }
+    }
+  else if (m_RegionContributors[i].size() == 1) //just one tile
+    {
+    SizeValueType tileInd = *m_RegionContributors[i].begin();
+    auto input = static_cast<const ImageType*>(this->GetInput(tileInd));
+    
+    if (!interpolate)
+      {
+      OffsetType tileToRegion = m_Regions[i].GetIndex() - m_InputMappings[tileInd].GetIndex();
+      RegionType iReg = m_Regions[i];
+      iReg.SetIndex(input->GetRequestedRegion().GetIndex() + tileToRegion);
+      ImageAlgorithm::Copy(input, m_SingleImage.GetPointer(), iReg, m_Regions[i]);
       }
-    else if (regionContributors[i].size() == 1) //just one tile
+    else
       {
-      SizeValueType tileInd = *regionContributors[i].begin();
-      auto input = static_cast<const ImageType*>(this->GetInput(tileInd));
-      
-      if (!interpolate)
+      Vector<typename ContinuousIndexType::ValueType, ImageDimension> ciDiff;
+      for (unsigned d = 0; d < ImageDimension; d++)
         {
-        OffsetType tileToRegion = regions[i].GetIndex() - inputMappings[tileInd].GetIndex();
-        RegionType iReg = regions[i];
-        iReg.SetIndex(input->GetRequestedRegion().GetIndex() + tileToRegion);
-        ImageAlgorithm::Copy(input, result.GetPointer(), iReg, regions[i]);
+        ciDiff[d] = input->GetRequestedRegion().GetIndex(d)
+            - m_InputsContinuousIndices[tileInd][d];
         }
-      else
+      typename TInterpolator::Pointer interp = TInterpolator::New();
+      interp->SetInputImage(input);
+      while (!oIt.IsAtEnd())
         {
-        Vector<typename ContinuousIndexType::ValueType, ImageDimension> ciDiff;
-        for (unsigned d = 0; d < ImageDimension; d++)
-          {
-          ciDiff[d] = input->GetRequestedRegion().GetIndex(d) - inputCIs[tileInd][d];
-          }
-        typename TInterpolator::Pointer interp = TInterpolator::New();
-        interp->SetInputImage(input);
-        while (!oIt.IsAtEnd())
-          {
-          ContinuousIndexType cInd = oIt.GetIndex();
-          cInd += ciDiff;
-          oIt.Set(interp->EvaluateAtContinuousIndex(cInd));
-          ++oIt;
-          }
+        ContinuousIndexType cInd = oIt.GetIndex();
+        cInd += ciDiff;
```

Comment:
This variable is used only on these 3 consecutive lines, as a temporary. Since its type is spelled out in full, I think spelling it out also would be repetitive and distracting to the reader:
```C++
        ContinuousIndexType continuousIndex = oIt.GetIndex();
        continuousIndex += continuousIndexDifference;
        oIt.Set(interp->EvaluateAtContinuousIndex(continuousIndex ));
```

Reply:
Done!